File: pe-math/src/main/java/org/saar/maths/objects/Polygon.java
package org.saar.maths.objects;

import org.joml.Vector2f;
import org.saar.maths.utils.Vector2;

import java.util.ArrayList;
import java.util.List;

public class Polygon {

    private final List<Vector2f> vertices = new ArrayList<>();

    public void addVertex(float x, float y) {
        vertices.add(Vector2.of(x, y));
    }

    public List<Vector2f> getVertices() {
        return vertices;
    }
}

File: pe-math/src/main/kotlin/org/saar/maths/objects/Polygon.kt
package org.saar.maths.objects

import org.joml.Vector2f
import org.saar.maths.utils.Vector2.of

class Polygon {

    val vertices = mutableListOf<Vector2f>()

    fun addVertex(x: Float, y: Float) {
        this.vertices.add(of(x, y))
    }
}
